### PR TITLE
Add undeclared Azure settings, modify test to exercise them

### PR DESCRIPTION
**Pull request by nicktindall**
Declares some settings that were undeclared by the Azure repository plugin.

Specifically, this PR allows use of
- `repository.azure.http_client.event_loop_executor_thread_count`
- `repository.azure.http_client.max_open_connections`
- `repository.azure.http_client.connection_timeout`
- `repository.azure.http_client.connection_max_idle_time`

### CODE REVIEW

**Comment by nicktindall** [comment]
I added tests that two of the settings are effective when set, the remaining ones don't seem to be easily testable

File: modules/repository-azure/src/internalClusterTest/java/org/elasticsearch/repositories/azure/AzureBlobStoreRepositoryTests.java [hunk 1]
```diff
@@ -262,6 +276,16 @@ private boolean isPutBlockList(String request) {
         }
     }
 
+    public void testSettingsTakeEffect() {
+        AzureClientProvider azureClientProvider = internalCluster().getInstance(AzureClientProvider.class);
+        assertEquals(MAX_CONNECTION_SETTING.get(), azureClientProvider.getConnectionProvider().maxConnections());
+        ThreadPool nodeThreadPool = internalCluster().getInstance(ThreadPool.class);
+        assertEquals(
+            EVENT_LOOP_THREAD_COUNT_SETTING.get(),
+            ((EsThreadPoolExecutor) nodeThreadPool.executor(AzureRepositoryPlugin.NETTY_EVENT_LOOP_THREAD_POOL_NAME)).getMaximumPoolSize()
```
Comment (ywangd) [inline modules/repository-azure/src/internalClusterTest/java/org/elasticsearch/repositories/azure/AzureBlobStoreRepositoryTests.java:285]: Nit: can this be
```
nodeThreadPool.info(AzureRepositoryPlugin.NETTY_EVENT_LOOP_THREAD_POOL_NAME).max()
```
?

**Review by ywangd — APPROVED** [review]
LGTM

I think this should be labelled as `>bug` instead? It has a visible impact and also could use a doc update.

Being able to registering all these settings is already a test. So I am not too worried about testing whether the remaining two getting picked up by the connectionProvider.

File: modules/repository-azure/src/internalClusterTest/java/org/elasticsearch/repositories/azure/AzureBlobStoreRepositoryTests.java (hunk 1 above)
Comment (nicktindall) [reply modules/repository-azure/src/internalClusterTest/java/org/elasticsearch/repositories/azure/AzureBlobStoreRepositoryTests.java]: Yes, that's much nicer, thanks.

**Pull request merged**